integration/run_test.go

```diff
@@ -333,12 +348,11 @@ disable = true
 			stderrLogMonitor.Add("listener", listener(failureChannel))
 			defer stderrLogMonitor.Remove("listener")
 
-			// Wait with network disabled. At this point the snapshotter should be
-			// retrying.
-			time.Sleep(time.Duration(tt.config.networkDisableMsec) * time.Millisecond)
-
 			// Restore network access
 			if tt.config.networkDisableMsec > 0 {
+				// Wait with network disabled. At this point the snapshotter should be
+				// retrying.
+				time.Sleep(time.Duration(tt.config.networkDisableMsec) * time.Millisecond)
 				// Restore access to the registry and image
 				sh.X("iptables", "-D", "OUTPUT", "-d", registryHostIP, "-j", "DROP")
```

Comment:
Does skipping this make our tests order dependent? If this test doesn't run last, do the next tests not have access to the registry?

Reply:
No, since every test runs with a new container. But, I think this is a valid concern anyway, so I did a bit of refactoring to deterministically restore network access at the end of every test.